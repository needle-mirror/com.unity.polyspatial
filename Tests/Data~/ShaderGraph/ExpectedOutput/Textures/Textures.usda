#usda 1.0
(
    metersPerUnit = 1
    upAxis = "Y"
)

def "MaterialX"
{
    def "Materials"
    {
        def Material "USD_Textures"
        {
            asset inputs:_Texture2D = @placeholder.png@ (colorSpace = "srgb_texture")
            float4 inputs:TextureTransform_Texture2D = (1, 1, 0, 0)
            float4 inputs:TextureTransform_Texture2DAsset_8f14bfbef458429a8aa3c2828046fc2b_Out_0_Texture2D = (1, 1, 0, 0)
            asset inputs:_Texture2DAsset_8f14bfbef458429a8aa3c2828046fc2b_Out_0_Texture2D = @placeholder.png@ (colorSpace = "srgb_texture")
            asset inputs:_SampleTexture2D_1474f2d5073b4ac29398add84bb23240_Texture_1_Texture2D = @placeholder.png@ (colorSpace = "srgb_texture")
            float4 inputs:TextureTransform_SampleTexture2D_1474f2d5073b4ac29398add84bb23240_Texture_1_Texture2D = (1, 1, 0, 0)
            asset inputs:_Triplanar_866a5c0c75f84e4c9b25421f3988b9a3_Texture_1_Texture2D = @placeholder.png@ (colorSpace = "srgb_texture")
            float4 inputs:TextureTransform_Texture2DAsset_67eda6922a484f11826572a12c4ea406_Out_0_Texture2D = (1, 1, 0, 0)
            asset inputs:_Texture2DAsset_67eda6922a484f11826572a12c4ea406_Out_0_Texture2D = @placeholder.png@ (colorSpace = "srgb_texture")
            asset inputs:_GatherTexture2D_5d27c369b1834db8a33a504c3a706ff1_Texture_1_Texture2D = @placeholder.png@ (colorSpace = "srgb_texture")
            float4 inputs:TextureTransform_GatherTexture2D_5d27c369b1834db8a33a504c3a706ff1_Texture_1_Texture2D = (1, 1, 0, 0)
            float3 inputs:TextureSize_GatherTexture2D_5d27c369b1834db8a33a504c3a706ff1_Texture_1_Texture2D = (0, 0, 0)

            token outputs:mtlx:surface.connect = <realitykit_unlit.outputs:out>
            token outputs:realitykit:vertex.connect = <GeometryModifier.outputs:out>

            def Shader "realitykit_unlit"
            {
                uniform token info:id = "ND_realitykit_unlit_surfaceshader"
                bool inputs:applyPostProcessToneMap = 0
                color3f inputs:color.connect = <../_177.outputs:out>
                token outputs:out
            }

            def Shader "GeometryModifier"
            {
                uniform token info:id = "ND_realitykit_geometrymodifier_vertexshader"
                token outputs:out
            }

            def Shader "_177"
            {
                uniform token info:id = "ND_swizzle_vector4_color3"
                float4 inputs:in.connect = <../_87.outputs:out>
                string inputs:channels = "xyz"
                color3f outputs:out
            }

            def Shader "_87"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_88.outputs:out>
                float4 outputs:out
            }

            def Shader "_88"
            {
                uniform token info:id = "ND_mix_vector4"
                float4 inputs:bg.connect = <../_89.outputs:out>
                float4 inputs:fg.connect = <../_90.outputs:out>
                float inputs:mix.connect = <../_103.outputs:out>
                float4 outputs:out
            }

            def Shader "_89"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_176.outputs:out>
                float4 outputs:out
            }

            def Shader "_176"
            {
                uniform token info:id = "ND_swizzle_color4_vector4"
                color4f inputs:in.connect = <../_82.outputs:out>
                string inputs:channels = "rgba"
                float4 outputs:out
            }

            def Shader "_82"
            {
                uniform token info:id = "ND_triplanarprojection_color4"
                color4f inputs:default = (1, 1, 1, 1)
                string inputs:filtertype = "cubic"
                float3 inputs:position.connect = <../_83.outputs:out>
                float3 inputs:normal.connect = <../_85.outputs:out>
                asset inputs:filex.connect = <../_86.outputs:out>
                asset inputs:filey.connect = <../_86.outputs:out>
                asset inputs:filez.connect = <../_86.outputs:out>
                color4f outputs:out
            }

            def Shader "_83"
            {
                uniform token info:id = "ND_multiply_vector3FA"
                float3 inputs:in1.connect = <../_84.outputs:out>
                float inputs:in2 = 1
                float3 outputs:out
            }

            def Shader "_84"
            {
                uniform token info:id = "ND_position_vector3"
                string inputs:space = "world"
                float3 outputs:out
            }

            def Shader "_90"
            {
                uniform token info:id = "ND_add_vector4"
                float4 inputs:in1.connect = <../_91.outputs:out>
                float4 inputs:in2.connect = <../_96.outputs:out>
                float4 outputs:out
            }

            def Shader "_96"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_97.outputs:out>
                float4 inputs:in2.connect = <../_98.outputs:out>
                float4 outputs:out
            }

            def Shader "_97"
            {
                uniform token info:id = "ND_subtract_vector4"
                float4 inputs:in1 = (1, 1, 1, 1)
                float4 inputs:in2.connect = <../_95.outputs:out>
                float4 outputs:out
            }

            def Shader "_95"
            {
                uniform token info:id = "ND_realitykit_step_vector4"
                float4 inputs:edge.connect = <../_89.outputs:out>
                float4 inputs:in = (0.5, 0.5, 0.5, 0.5)
                float4 outputs:out
            }

            def Shader "_91"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_92.outputs:out>
                float4 inputs:in2.connect = <../_95.outputs:out>
                float4 outputs:out
            }

            def Shader "_92"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_93.outputs:out>
                float4 inputs:in2.connect = <../_94.outputs:out>
                float4 outputs:out
            }

            def Shader "_94"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_75.outputs:out>
                float4 outputs:out
            }

            def Shader "_75"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_76.outputs:out>
                float4 outputs:out
            }

            def Shader "_76"
            {
                uniform token info:id = "ND_mix_vector4"
                float4 inputs:bg.connect = <../_77.outputs:out>
                float4 inputs:fg.connect = <../_78.outputs:out>
                float inputs:mix.connect = <../_81.outputs:out>
                float4 outputs:out
            }

            def Shader "_77"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_39.outputs:out>
                float4 outputs:out
            }

            def Shader "_39"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_40.outputs:out>
                float4 outputs:out
            }

            def Shader "_40"
            {
                uniform token info:id = "ND_mix_vector4"
                float4 inputs:bg.connect = <../_41.outputs:out>
                float4 inputs:fg.connect = <../_42.outputs:out>
                float inputs:mix.connect = <../_55.outputs:out>
                float4 outputs:out
            }

            def Shader "_41"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_1.outputs:out>
                float4 outputs:out
            }

            def Shader "_1"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_2.outputs:out>
                float4 outputs:out
            }

            def Shader "_2"
            {
                uniform token info:id = "ND_RealityKitTexture2D_vector4"
                asset inputs:file.connect = <../_3.outputs:out>
                float2 inputs:texcoord.connect = <../_4.outputs:out>
                string inputs:min_filter = "linear"
                string inputs:mag_filter = "linear"
                string inputs:mip_filter = "nearest"
                string inputs:u_wrap_mode = "repeat"
                string inputs:v_wrap_mode = "repeat"
                int inputs:max_anisotropy = 1
                float4 outputs:out
            }

            def Shader "_3"
            {
                uniform token info:id = "ND_dot_filename"
                asset inputs:in.connect = <../.inputs:_Texture2DAsset_8f14bfbef458429a8aa3c2828046fc2b_Out_0_Texture2D>
                asset outputs:out
            }

            def Shader "_93"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1 = (2, 2, 2, 2)
                float4 inputs:in2.connect = <../_89.outputs:out>
                float4 outputs:out
            }

            def Shader "_103"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_166.outputs:out>
                float outputs:out
            }

            def Shader "_166"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in.connect = <../_167.outputs:out>
                float outputs:out
            }

            def Shader "_167"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../_124.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_124"
            {
                uniform token info:id = "ND_combine4_vector4"
                float inputs:in1.connect = <../_125.outputs:out>
                float inputs:in2.connect = <../_148.outputs:out>
                float inputs:in3.connect = <../_154.outputs:out>
                float inputs:in4.connect = <../_160.outputs:out>
                float4 outputs:out
            }

            def Shader "_154"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../_155.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_155"
            {
                uniform token info:id = "ND_RealityKitTexture2DLOD_vector4"
                asset inputs:file.connect = <../_127.outputs:out>
                float2 inputs:texcoord.connect = <../_156.outputs:out>
                string inputs:min_filter = "linear"
                string inputs:mag_filter = "linear"
                string inputs:mip_filter = "nearest"
                string inputs:u_wrap_mode = "repeat"
                string inputs:v_wrap_mode = "repeat"
                int inputs:max_anisotropy = 1
                float4 outputs:out
            }

            def Shader "_127"
            {
                uniform token info:id = "ND_dot_filename"
                asset inputs:in.connect = <../.inputs:_GatherTexture2D_5d27c369b1834db8a33a504c3a706ff1_Texture_1_Texture2D>
                asset outputs:out
            }

            def Shader "_148"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../_149.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_149"
            {
                uniform token info:id = "ND_RealityKitTexture2DLOD_vector4"
                asset inputs:file.connect = <../_127.outputs:out>
                float2 inputs:texcoord.connect = <../_150.outputs:out>
                string inputs:min_filter = "linear"
                string inputs:mag_filter = "linear"
                string inputs:mip_filter = "nearest"
                string inputs:u_wrap_mode = "repeat"
                string inputs:v_wrap_mode = "repeat"
                int inputs:max_anisotropy = 1
                float4 outputs:out
            }

            def Shader "_150"
            {
                uniform token info:id = "ND_divide_vector2"
                float2 inputs:in1.connect = <../_151.outputs:out>
                float2 inputs:in2.connect = <../_141.outputs:out>
                float2 outputs:out
            }

            def Shader "_141"
            {
                uniform token info:id = "ND_swizzle_vector3_vector2"
                float3 inputs:in.connect = <../.inputs:TextureSize_GatherTexture2D_5d27c369b1834db8a33a504c3a706ff1_Texture_1_Texture2D>
                string inputs:channels = "xy"
                float2 outputs:out
            }

            def Shader "_151"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_152.outputs:out>
                float2 inputs:in2.connect = <../_142.outputs:out>
                float2 outputs:out
            }

            def Shader "_142"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_143.outputs:out>
                float2 inputs:in2 = (0.5, 0.5)
                float2 outputs:out
            }

            def Shader "_143"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_144.outputs:out>
                float2 inputs:in2.connect = <../_147.outputs:out>
                float2 outputs:out
            }

            def Shader "_144"
            {
                uniform token info:id = "ND_floor_vector2"
                float2 inputs:in.connect = <../_145.outputs:out>
                float2 outputs:out
            }

            def Shader "_145"
            {
                uniform token info:id = "ND_absval_vector2"
                float2 inputs:in.connect = <../_146.outputs:out>
                float2 outputs:out
            }

            def Shader "_146"
            {
                uniform token info:id = "ND_dot_vector2"
                float2 inputs:in = (0, 0)
                float2 outputs:out
            }

            def Shader "_152"
            {
                uniform token info:id = "ND_floor_vector2"
                float2 inputs:in.connect = <../_153.outputs:out>
                float2 outputs:out
            }

            def Shader "_153"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_132.outputs:out>
                float2 inputs:in2 = (0.5, 0.5)
                float2 outputs:out
            }

            def Shader "_132"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_133.outputs:out>
                float2 inputs:in2.connect = <../_141.outputs:out>
                float2 outputs:out
            }

            def Shader "_133"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_134.outputs:out>
                float2 inputs:in2.connect = <../_140.outputs:out>
                float2 outputs:out
            }

            def Shader "_140"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_GatherTexture2D_5d27c369b1834db8a33a504c3a706ff1_Texture_1_Texture2D>
                string inputs:channels = "zw"
                float2 outputs:out
            }

            def Shader "_134"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_135.outputs:out>
                float2 inputs:in2.connect = <../_139.outputs:out>
                float2 outputs:out
            }

            def Shader "_139"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_GatherTexture2D_5d27c369b1834db8a33a504c3a706ff1_Texture_1_Texture2D>
                string inputs:channels = "xy"
                float2 outputs:out
            }

            def Shader "_135"
            {
                uniform token info:id = "ND_dot_vector2"
                float2 inputs:in.connect = <../_138.outputs:out>
                float2 outputs:out
            }

            def Shader "_138"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_137.outputs:out>
                float2 inputs:in2 = (0, 1)
                float2 outputs:out
            }

            def Shader "_137"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_136.outputs:out>
                float2 inputs:in2 = (1, -1)
                float2 outputs:out
            }

            def Shader "_136"
            {
                uniform token info:id = "ND_texcoord_vector2"
                int inputs:index = 0
                float2 outputs:out
            }

            def Shader "_125"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../_126.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_126"
            {
                uniform token info:id = "ND_RealityKitTexture2DLOD_vector4"
                asset inputs:file.connect = <../_127.outputs:out>
                float2 inputs:texcoord.connect = <../_128.outputs:out>
                string inputs:min_filter = "linear"
                string inputs:mag_filter = "linear"
                string inputs:mip_filter = "nearest"
                string inputs:u_wrap_mode = "repeat"
                string inputs:v_wrap_mode = "repeat"
                int inputs:max_anisotropy = 1
                float4 outputs:out
            }

            def Shader "_128"
            {
                uniform token info:id = "ND_divide_vector2"
                float2 inputs:in1.connect = <../_129.outputs:out>
                float2 inputs:in2.connect = <../_141.outputs:out>
                float2 outputs:out
            }

            def Shader "_129"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_130.outputs:out>
                float2 inputs:in2.connect = <../_142.outputs:out>
                float2 outputs:out
            }

            def Shader "_130"
            {
                uniform token info:id = "ND_floor_vector2"
                float2 inputs:in.connect = <../_131.outputs:out>
                float2 outputs:out
            }

            def Shader "_131"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_132.outputs:out>
                float2 inputs:in2 = (-0.5, 0.5)
                float2 outputs:out
            }

            def Shader "_85"
            {
                uniform token info:id = "ND_normal_vector3"
                string inputs:space = "world"
                float3 outputs:out
            }

            def Shader "_86"
            {
                uniform token info:id = "ND_constant_filename"
                asset inputs:value.connect = <../.inputs:_Triplanar_866a5c0c75f84e4c9b25421f3988b9a3_Texture_1_Texture2D>
                asset outputs:out
            }

            def Shader "_98"
            {
                uniform token info:id = "ND_subtract_vector4"
                float4 inputs:in1 = (1, 1, 1, 1)
                float4 inputs:in2.connect = <../_99.outputs:out>
                float4 outputs:out
            }

            def Shader "_99"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_100.outputs:out>
                float4 inputs:in2.connect = <../_102.outputs:out>
                float4 outputs:out
            }

            def Shader "_102"
            {
                uniform token info:id = "ND_subtract_vector4"
                float4 inputs:in1 = (1, 1, 1, 1)
                float4 inputs:in2.connect = <../_94.outputs:out>
                float4 outputs:out
            }

            def Shader "_100"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1 = (2, 2, 2, 2)
                float4 inputs:in2.connect = <../_101.outputs:out>
                float4 outputs:out
            }

            def Shader "_101"
            {
                uniform token info:id = "ND_subtract_vector4"
                float4 inputs:in1 = (1, 1, 1, 1)
                float4 inputs:in2.connect = <../_89.outputs:out>
                float4 outputs:out
            }

            def Shader "_78"
            {
                uniform token info:id = "ND_absval_vector4"
                float4 inputs:in.connect = <../_79.outputs:out>
                float4 outputs:out
            }

            def Shader "_79"
            {
                uniform token info:id = "ND_subtract_vector4"
                float4 inputs:in1.connect = <../_80.outputs:out>
                float4 inputs:in2.connect = <../_77.outputs:out>
                float4 outputs:out
            }

            def Shader "_80"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_56.outputs:out>
                float4 outputs:out
            }

            def Shader "_56"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_57.outputs:out>
                float4 outputs:out
            }

            def Shader "_57"
            {
                uniform token info:id = "ND_RealityKitTexture2D_vector4"
                asset inputs:file.connect = <../_58.outputs:out>
                float2 inputs:texcoord.connect = <../_59.outputs:out>
                string inputs:min_filter = "linear"
                string inputs:mag_filter = "linear"
                string inputs:mip_filter = "nearest"
                string inputs:u_wrap_mode = "repeat"
                string inputs:v_wrap_mode = "repeat"
                int inputs:max_anisotropy = 1
                float4 outputs:out
            }

            def Shader "_59"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_60.outputs:out>
                float2 inputs:in2.connect = <../_66.outputs:out>
                float2 outputs:out
            }

            def Shader "_66"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_SampleTexture2D_1474f2d5073b4ac29398add84bb23240_Texture_1_Texture2D>
                string inputs:channels = "zw"
                float2 outputs:out
            }

            def Shader "_60"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_61.outputs:out>
                float2 inputs:in2.connect = <../_65.outputs:out>
                float2 outputs:out
            }

            def Shader "_65"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_SampleTexture2D_1474f2d5073b4ac29398add84bb23240_Texture_1_Texture2D>
                string inputs:channels = "xy"
                float2 outputs:out
            }

            def Shader "_61"
            {
                uniform token info:id = "ND_dot_vector2"
                float2 inputs:in.connect = <../_64.outputs:out>
                float2 outputs:out
            }

            def Shader "_64"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_63.outputs:out>
                float2 inputs:in2 = (0, 1)
                float2 outputs:out
            }

            def Shader "_63"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_62.outputs:out>
                float2 inputs:in2 = (1, -1)
                float2 outputs:out
            }

            def Shader "_62"
            {
                uniform token info:id = "ND_texcoord_vector2"
                int inputs:index = 0
                float2 outputs:out
            }

            def Shader "_58"
            {
                uniform token info:id = "ND_dot_filename"
                asset inputs:in.connect = <../.inputs:_SampleTexture2D_1474f2d5073b4ac29398add84bb23240_Texture_1_Texture2D>
                asset outputs:out
            }

            def Shader "_81"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in = 1
                float outputs:out
            }

            def Shader "_42"
            {
                uniform token info:id = "ND_add_vector4"
                float4 inputs:in1.connect = <../_43.outputs:out>
                float4 inputs:in2.connect = <../_48.outputs:out>
                float4 outputs:out
            }

            def Shader "_48"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_49.outputs:out>
                float4 inputs:in2.connect = <../_50.outputs:out>
                float4 outputs:out
            }

            def Shader "_50"
            {
                uniform token info:id = "ND_subtract_vector4"
                float4 inputs:in1 = (1, 1, 1, 1)
                float4 inputs:in2.connect = <../_51.outputs:out>
                float4 outputs:out
            }

            def Shader "_51"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_52.outputs:out>
                float4 inputs:in2.connect = <../_54.outputs:out>
                float4 outputs:out
            }

            def Shader "_54"
            {
                uniform token info:id = "ND_subtract_vector4"
                float4 inputs:in1 = (1, 1, 1, 1)
                float4 inputs:in2.connect = <../_46.outputs:out>
                float4 outputs:out
            }

            def Shader "_46"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_20.outputs:out>
                float4 outputs:out
            }

            def Shader "_20"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_21.outputs:out>
                float4 outputs:out
            }

            def Shader "_21"
            {
                uniform token info:id = "ND_RealityKitTexture2D_vector4"
                asset inputs:file.connect = <../_22.outputs:out>
                float2 inputs:texcoord.connect = <../_23.outputs:out>
                string inputs:min_filter = "linear"
                string inputs:mag_filter = "linear"
                string inputs:mip_filter = "nearest"
                string inputs:u_wrap_mode = "repeat"
                string inputs:v_wrap_mode = "repeat"
                int inputs:max_anisotropy = 1
                float4 outputs:out
            }

            def Shader "_23"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_24.outputs:out>
                float2 inputs:in2.connect = <../_30.outputs:out>
                float2 outputs:out
            }

            def Shader "_30"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_Texture2D>
                string inputs:channels = "zw"
                float2 outputs:out
            }

            def Shader "_24"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_25.outputs:out>
                float2 inputs:in2.connect = <../_29.outputs:out>
                float2 outputs:out
            }

            def Shader "_29"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_Texture2D>
                string inputs:channels = "xy"
                float2 outputs:out
            }

            def Shader "_25"
            {
                uniform token info:id = "ND_dot_vector2"
                float2 inputs:in.connect = <../_28.outputs:out>
                float2 outputs:out
            }

            def Shader "_28"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_27.outputs:out>
                float2 inputs:in2 = (0, 1)
                float2 outputs:out
            }

            def Shader "_27"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_26.outputs:out>
                float2 inputs:in2 = (1, -1)
                float2 outputs:out
            }

            def Shader "_26"
            {
                uniform token info:id = "ND_texcoord_vector2"
                int inputs:index = 0
                float2 outputs:out
            }

            def Shader "_22"
            {
                uniform token info:id = "ND_dot_filename"
                asset inputs:in.connect = <../.inputs:_Texture2D>
                asset outputs:out
            }

            def Shader "_52"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1 = (2, 2, 2, 2)
                float4 inputs:in2.connect = <../_53.outputs:out>
                float4 outputs:out
            }

            def Shader "_53"
            {
                uniform token info:id = "ND_subtract_vector4"
                float4 inputs:in1 = (1, 1, 1, 1)
                float4 inputs:in2.connect = <../_41.outputs:out>
                float4 outputs:out
            }

            def Shader "_49"
            {
                uniform token info:id = "ND_subtract_vector4"
                float4 inputs:in1 = (1, 1, 1, 1)
                float4 inputs:in2.connect = <../_47.outputs:out>
                float4 outputs:out
            }

            def Shader "_47"
            {
                uniform token info:id = "ND_realitykit_step_vector4"
                float4 inputs:edge.connect = <../_41.outputs:out>
                float4 inputs:in = (0.5, 0.5, 0.5, 0.5)
                float4 outputs:out
            }

            def Shader "_43"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_44.outputs:out>
                float4 inputs:in2.connect = <../_47.outputs:out>
                float4 outputs:out
            }

            def Shader "_44"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_45.outputs:out>
                float4 inputs:in2.connect = <../_46.outputs:out>
                float4 outputs:out
            }

            def Shader "_45"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1 = (2, 2, 2, 2)
                float4 inputs:in2.connect = <../_41.outputs:out>
                float4 outputs:out
            }

            def Shader "_55"
            {
                uniform token info:id = "ND_dot_float"
                float inputs:in = 1
                float outputs:out
            }

            def Shader "_4"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_5.outputs:out>
                float2 inputs:in2.connect = <../_11.outputs:out>
                float2 outputs:out
            }

            def Shader "_11"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_Texture2DAsset_8f14bfbef458429a8aa3c2828046fc2b_Out_0_Texture2D>
                string inputs:channels = "zw"
                float2 outputs:out
            }

            def Shader "_5"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_6.outputs:out>
                float2 inputs:in2.connect = <../_10.outputs:out>
                float2 outputs:out
            }

            def Shader "_10"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_Texture2DAsset_8f14bfbef458429a8aa3c2828046fc2b_Out_0_Texture2D>
                string inputs:channels = "xy"
                float2 outputs:out
            }

            def Shader "_6"
            {
                uniform token info:id = "ND_dot_vector2"
                float2 inputs:in.connect = <../_9.outputs:out>
                float2 outputs:out
            }

            def Shader "_9"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_8.outputs:out>
                float2 inputs:in2 = (0, 1)
                float2 outputs:out
            }

            def Shader "_8"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_7.outputs:out>
                float2 inputs:in2 = (1, -1)
                float2 outputs:out
            }

            def Shader "_7"
            {
                uniform token info:id = "ND_texcoord_vector2"
                int inputs:index = 0
                float2 outputs:out
            }

            def Shader "_160"
            {
                uniform token info:id = "ND_swizzle_vector4_float"
                float4 inputs:in.connect = <../_161.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_161"
            {
                uniform token info:id = "ND_RealityKitTexture2DLOD_vector4"
                asset inputs:file.connect = <../_127.outputs:out>
                float2 inputs:texcoord.connect = <../_162.outputs:out>
                string inputs:min_filter = "linear"
                string inputs:mag_filter = "linear"
                string inputs:mip_filter = "nearest"
                string inputs:u_wrap_mode = "repeat"
                string inputs:v_wrap_mode = "repeat"
                int inputs:max_anisotropy = 1
                float4 outputs:out
            }

            def Shader "_162"
            {
                uniform token info:id = "ND_divide_vector2"
                float2 inputs:in1.connect = <../_163.outputs:out>
                float2 inputs:in2.connect = <../_141.outputs:out>
                float2 outputs:out
            }

            def Shader "_163"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_164.outputs:out>
                float2 inputs:in2.connect = <../_142.outputs:out>
                float2 outputs:out
            }

            def Shader "_164"
            {
                uniform token info:id = "ND_floor_vector2"
                float2 inputs:in.connect = <../_165.outputs:out>
                float2 outputs:out
            }

            def Shader "_165"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_132.outputs:out>
                float2 inputs:in2 = (-0.5, -0.5)
                float2 outputs:out
            }

            def Shader "_156"
            {
                uniform token info:id = "ND_divide_vector2"
                float2 inputs:in1.connect = <../_157.outputs:out>
                float2 inputs:in2.connect = <../_141.outputs:out>
                float2 outputs:out
            }

            def Shader "_157"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_158.outputs:out>
                float2 inputs:in2.connect = <../_142.outputs:out>
                float2 outputs:out
            }

            def Shader "_158"
            {
                uniform token info:id = "ND_floor_vector2"
                float2 inputs:in.connect = <../_159.outputs:out>
                float2 outputs:out
            }

            def Shader "_159"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_132.outputs:out>
                float2 inputs:in2 = (0.5, -0.5)
                float2 outputs:out
            }

            def Shader "_147"
            {
                uniform token info:id = "ND_sign_vector2"
                float2 inputs:in.connect = <../_146.outputs:out>
                float2 outputs:out
            }
        }
    }
}
