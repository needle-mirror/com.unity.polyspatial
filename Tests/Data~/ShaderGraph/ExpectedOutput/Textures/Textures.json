{"properties":[{"name":"_Texture2D","type":6},{"name":"TextureTransform_Texture2D","type":3},{"name":"TextureTransform_Texture2DAsset_8f14bfbef458429a8aa3c2828046fc2b_Out_0_Texture2D","type":3},{"name":"_Texture2DAsset_8f14bfbef458429a8aa3c2828046fc2b_Out_0_Texture2D","type":6},{"name":"_SampleTexture2D_1474f2d5073b4ac29398add84bb23240_Texture_1_Texture2D","type":6},{"name":"TextureTransform_SampleTexture2D_1474f2d5073b4ac29398add84bb23240_Texture_1_Texture2D","type":3},{"name":"_Triplanar_866a5c0c75f84e4c9b25421f3988b9a3_Texture_1_Texture2D","type":6},{"name":"TextureTransform_Texture2DAsset_67eda6922a484f11826572a12c4ea406_Out_0_Texture2D","type":3},{"name":"_Texture2DAsset_67eda6922a484f11826572a12c4ea406_Out_0_Texture2D","type":6},{"name":"_GatherTexture2D_5d27c369b1834db8a33a504c3a706ff1_Texture_1_Texture2D","type":6},{"name":"TextureTransform_GatherTexture2D_5d27c369b1834db8a33a504c3a706ff1_Texture_1_Texture2D","type":3},{"name":"TextureSize_GatherTexture2D_5d27c369b1834db8a33a504c3a706ff1_Texture_1_Texture2D","type":3}]}